### PR TITLE
dotCMS/core#22904 Block Editor: Add char count information

### LINKED ISSUE #22904: Block Editor: Add char count information
**Is your feature request related to a problem? Please describe.**
As a content author I want to know how long is the content I'm creating, this can help to optimize my content for SEO and improve performance.

**Describe the solution you'd like**
We can use the existing extension: https://tiptap.dev/api/extensions/character-count

**Describe alternatives you've considered**
N/A

**Additional context**
- How do we count contentlets and images?
- Where do we show this?
- Should we have a field var to enable and disable?

### PR DESCRIPTION
**Pull request by rjvelazco**
**What we did**

1. Add `chartCount` tiptap extension.
2. Calculate estimated reading time.
3. Add `charLimit` field variable.
4. Add `displayCountBar` field variable to control whether character count and estimated reading time are displayed.
5. Update de JSON Object to store `attr`.

**New Document JSON**

```JSON
{
    "type": "doc",
    "attrs": {
        "chartCount": 11,
        "readingTime": 1
    },
    "content": [{}]
}
```

Now we have `chartCount` and `readingTime` (Estimated time in minutes).

### CODE REVIEW

File: core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.ts [hunk 1]
```diff
@@ -49,20 +56,50 @@ export class DotBlockEditorComponent implements OnInit {
...
 
     constructor(private injector: Injector, public viewContainerRef: ViewContainerRef) {}
 
     ngOnInit() {
         this.editor = new Editor({
             extensions: this.setEditorExtensions()
         });
+
+        this.editor.on('create', () => this.updateChartCount());
+        this.subject.pipe(debounceTime(250)).subscribe(() => this.updateChartCount());
```
Comment (fmontes) [inline core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.ts:95]: takeUntil?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.ts (hunk 1 above)
Comment (rjvelazco) [reply core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.ts:95]: Right! Great catch!

File: core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.scss [hunk 2]
```diff
@@ -26,6 +26,24 @@
         outline: $gray-light solid 1px;
     }
 
+    .count-bar_container {
+        display: flex;
+        justify-content: flex-end;
+        align-items: flex-end;
+        color: #868e96;
```
[the remark this answers is not in the document]
Comment (rjvelazco) [reply core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.scss:33]: Sure!

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_field.jsp [hunk 3]
```diff
@@ -181,6 +184,12 @@
                 if("allowedBlocks".equalsIgnoreCase(fv.getKey())){
                     allowedBlocks = fv.getValue().replaceAll("[^a-zA-Z0-9,]", "");
                 }
+                if("displayCountBar".equalsIgnoreCase(fv.getKey())){
```
Comment (hmoreras) [inline dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_field.jsp:187]: add the doc needed in the card.

**Review by hmoreras — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**